lib/bundler/definition.rb

```diff
@@ -545,6 +539,20 @@ def unlocking?
 
   private
 
+    def add_platforms
+      (@dependencies.flat_map(&:expanded_platforms) + current_platforms).uniq.each do |platform|
+        add_platform(platform)
+      end
+    end
+
+    def current_platforms
+      current_platform = Bundler.local_platform
+      platforms = []
```

Comment:
What about to use `tap` in here?

```ruby
[].tap do |platforms|
  platforms << current_platform if Bundler.feature_flag.specific_platform?
  platforms << generic(current_platform)
end
```

Reply:
Soungs good, yeah. I'll update it.

Reply:
Done in https://github.com/bundler/bundler/pull/7215/commits/d7b5cf1fcf5cd86cd579450414cb9ff5ab03cf5e.